src/main/java/lambdastreams/task2/Book.java

```diff
@@ -0,0 +1,116 @@
+package lambdastreams.task2;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+
+public class Book {
+
+    private String name;
+    private Integer price;
+
+    public Book(String name, Integer price) {
+        this.name = name;
+        this.price = price;
+    }
+
+    public Book(String name){
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Integer getPrice() {
+        return price;
+    }
+
+    public static List<Book> createListOfBooks(Integer booksQuantity) {
+        List<Book> listOfBooks = new ArrayList<>();
+        for (int i = 0; i < booksQuantity; i++) {
+            String randomBookName = RandomStringUtils.randomAlphabetic(3, 10).toLowerCase(Locale.ROOT);
+            randomBookName = StringUtils.capitalize(randomBookName);
+            Integer randomBookPrice = new Random().nextInt(100);
+            Book book = new Book(randomBookName, randomBookPrice);
+            listOfBooks.add(book);
+        }
+        return listOfBooks;
+    }
+
+    public static List<String> createListOfNamesOfAllBooks(List<Book> listOfBooks) {
+        return listOfBooks.stream()
+                .map(Book::getName)
+                .collect(Collectors.toList());
+    }
+
+    public static int totalPriceOfAllBooks(List<Book> listOfBooks) {
+        return listOfBooks.stream()
+                .map(Book::getPrice)
+                .reduce(0, Integer::sum);
+    }
+
+    public static List<String> createListOfBooksContainsA(List<Book> listOfBooks) {
+        return listOfBooks.stream()
+                .map(Book::getName)
+                .filter(name -> name.contains("A") | name.contains("a"))
```

Comment:
"|" - is a bitwise operator in Java, please change to logical operator "||"

Reply:
Done